src/grinder/views.py

```diff
@@ -1,7 +1,183 @@
+    return render(request, 'sign_up.html', context={'form': form})
+
+
+def GetUser(request):
+    if request.method == 'POST':
+        # gets user from db if valid and updates if needed access & refresh tokens
+        # First we get username and password to auth spotify-grinder user
+        user_name = request.POST.get('username')
+        password = request.POST.get('password')
+        print(user_name, password)
+        user = authenticate(username=user_name, password=password)
+        if user is not None:
+            # A backend authenticated the credentials
+            # Is the account active? It could have been disabled.
+            if user.is_active:
+                # If the account is valid and active, we can log the user in.
+                # We'll send the user back to the homepage.
+                login(request, user)
+                # Auth Step 1: Authorization
+                url_args = "&".join(["{}={}".format(key, quote(val)) for key, val in auth_query_parameters.items()])
+                auth_url = "{}/?{}".format(SPOTIFY_AUTH_URL, url_args)
+                return redirect(auth_url)
+            else:
+                # An inactive account was used - no logging in!
+                return HttpResponse("Your spotify-grinder account is disabled.")
+        else:
+             # No backend authenticated the credentials
+             return HttpResponse("This spotify-grinder account is not valid.")
+    else:
+        return render(request, 'sign_in.html', context={})
+
+# Auth step 2: User login
+
+# Auth step 3: Spotify redirects the User to auth
+
+@login_required(login_url='sign_in/')
+def auth(request):
+    # Auth Step 4: Requests refresh and access tokens
+    auth_token = request.GET.get('code')
+    #user.spRefresh()
+    code_payload = {
+        "grant_type": "authorization_code",
+        "code": str(auth_token),
+        "redirect_uri": REDIRECT_URI,
+        'client_id': CLIENT_ID,
+        'client_secret': CLIENT_SECRET,
+    }
+    post_request = requests.post(SPOTIFY_TOKEN_URL, data=code_payload)
+
+    # Auth Step 5: Tokens are Returned to Application
+    response_data = json.loads(post_request.text)
+    access_token = response_data["access_token"]
+    refresh_token = response_data["refresh_token"]
+    token_type = response_data["token_type"]
+    expires_in = response_data["expires_in"]
+
+    # Updating the user.access_token
+    user = User.objects.get(username=request.user.username)
+
+    # Update User tokens and expire date
+    user.access_token = access_token
+    user.refresh_token = refresh_token
+    user.expires_in = expires_in
+
+    # Auth Step 6: Use the access token to access Spotify API
+    authorization_header = {"Authorization": "Bearer {}".format(access_token)}
+
+    # Get profile data
+    Current_users_profile_APIendpoint = "{}/me".format(SPOTIFY_API_URL)
+    User_profile_APIendpoint = "{}/users/{}".format(SPOTIFY_API_URL, user.username)
```

Comment:
can we set these as objects in the class, that basically gets populated once we obtain these information. Seems like only `authorization_header` that needs an obtainable value also the username in `User_profile_APIendpoint`
so maybe move them to a function that if you passed a value it sets it and if you didn't it gives you whats in it